pkg/util/github/github.go

```diff
@@ -120,13 +121,14 @@ func DownloadOpenShiftReleaseBinary(binaryType OpenShiftBinaryType, osType minis
 
 	// Download the asset
 	var asset io.Reader
-	asset, url, err := client.Repositories.DownloadReleaseAsset("openshift", "origin", assetID)
+	asset, uri, err := client.Repositories.DownloadReleaseAsset("openshift", "origin", assetID)
 	if err != nil {
 		return errors.Wrap(err, fmt.Sprintf("Cannot download OpenShift release asset %d", assetID))
 	}
-	if len(url) > 0 {
+	if len(uri) > 0 {
 		fmt.Println(fmt.Sprintf("Downloading OpenShift binary '%s' version '%s'", binaryType.String(), *release.TagName))
-		httpResp, err := http.Get(url)
+		myClient := util.GetHttpClient(proxyUrl)
```

Comment:
Ah, I see what you did.
For some reason, I do think this code is more complex than it needs to be. Readability is not very nice...